GroupRepresentationsForCAP/gap/SemisimpleCategoryMorphisms.gi

```diff
@@ -329,3 +329,72 @@ InstallMethod( Display,
     
 end );
 
+##
+InstallMethod( LaTeXStringOp,
+        "for a morphism in a semisimple category",
+        [ IsSemisimpleCategoryMorphism ],
+        
+  function ( morphism )
+    local morphism_list, source_object_list, range_object_list, source_object_string, range_object_string, diag_entries, matrix, string, elem;
+    
+    morphism_list := SemisimpleCategoryMorphismList( morphism );
+    
+    morphism_list := Filtered( morphism_list, pair -> not( IsZeroForObjects( Source( pair[1] ) ) or IsZeroForObjects( Range( pair[1] ) ) ) );
+    
+    if IsEmpty( morphism_list ) then
+        
+        Display( "0" );
+        
+    else
+        
+        source_object_list := List( morphism_list, m -> [ NrRows( UnderlyingMatrix( m[1] ) ), m[2] ] );
+        range_object_list := List( morphism_list, m -> [ NrCols( UnderlyingMatrix( m[1] ) ), m[2] ] );
+        
+        source_object_string := LaTeXStringOfSemisimpleCategoryObjectList( source_object_list );
+        range_object_string := LaTeXStringOfSemisimpleCategoryObjectList( range_object_list );
+        
+        diag_entries := [ ];
+        
+        for elem in morphism_list do
+            
+            matrix := UnderlyingMatrix( elem[1] );
+            
+            if NrRows( matrix ) = 1 and NrCols( matrix ) = 1 then
+                
+                Add( diag_entries, LaTeXStringOp( matrix[1, 1] ) );
+                
+            else
+                
+                Add( diag_entries, LaTeXStringOp( matrix ) );
+                
+            fi;
+            
+        od;
+        
+        string := "";
+        Append( string, source_object_string );
+        Append( string, " \\xrightarrow{\\operatorname{diag}(" );
+        Append( string, JoinStringsWithSeparator( diag_entries, "," ) );
+        Append( string, ")} " );
+        Append( string, range_object_string );
+        
+        return string;
+        
+    fi;
+    
+end );
+
+##
+InstallMethod( IsShowable,
```

Comment:
This looks to me like a generic function that you always want to be installed whenever you want a GAP object to be showable (you install exactly the same function for `IsSemisimpleCategoryObject` below). Maybe that should be taken as a hint for the LaTeX interface that you implicitly propagate with IsShowable/LaTeXStringOp at the moment. I would suggest to outsource this code to a function within `ToolsForHomalg` called something like `ActivateIsShowable( filter )`.

Reply:
You are correct that right now this is rather generic, since we know that we can always generate sensible LaTeX strings. However, in more complicated cases I expect that there will be further checks, for example if we only want to support LaTeX output if the underlying category supports LaTeX output, or only if the output is reasonable small. So I am not sure if a general function `ActivateIsShowable( mime_type, filter )` would be widely used. But of course it would not hurt introducing such a function.

@mohamed-barakat Do you have an opinion on this?

Reply:
I have created a PR: homalg-project/homalg_project#353

Reply:
Thanks, merged.